files/aiida-core-constraints.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+"""Script to convert aiida setup.json to pip constraints file.
+
+Usage::
+
+    python aiida-core-constraints.py path/to/setup.json extras1 extras2 ...
+"""
+import json
+from pathlib import Path
+import sys
+
+def remove_extra(req):
+    """Extras are not allowed in constraints files, e.g. dep[extra]>0.1 -> dep>0.1 """
```

Comment:
why is that?

Reply:
see here https://pip.pypa.io/en/latest/user_guide/#changes-to-the-pip-dependency-resolver-in-20-2-2020, which links to https://github.com/pypa/pip/issues/6628

Reply:
Ah, fair enough. Using pip-tools would fix that too.